util/conversion/conversion.go

```diff
@@ -94,7 +95,18 @@ func ConvertReferenceAPIContract(ctx context.Context, c client.Client, ref *core
 
 // MarshalData stores the source object as json data in the destination object annotations map.
 func MarshalData(src metav1.Object, dst metav1.Object) error {
-	data, err := json.Marshal(src)
+	u := &unstructured.Unstructured{}
+	scheme, err := clusterv1.SchemeBuilder.Build()
+	if err != nil {
+		return err
+	}
+	err = scheme.Convert(src, u, nil)
```

Comment:
Is `scheme.Convert` the only way to get to unstructured object? Is there another way to do this?

Reply:
No need to create a new scheme - you can use `runtime.DefaultUnstructuredConverter`.